apps/api/app/identity_api.py

```
@@ -0,0 +1,955 @@
+)
+def okta_sso_callback(
+    state: str,
+    session: SessionDep,
+    code: str | None = Query(default=None),
+    email: str | None = Query(default=None),
+    sub: str | None = Query(default=None),
+    groups: str | None = Query(default=None),
+    error: str | None = Query(default=None),
+) -> dict[str, Any]:
+    if error:
+        raise unauthorized("Okta authentication failed", details={"error": error})
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)